include/revng/Yield/ControlFlow/Graph.h

```diff
@@ -5,20 +5,44 @@
 //
 
 #include "revng/GraphLayout/Graphs.h"
+#include "revng/Pipeline/Location.h"
+#include "revng/Pipes/Ranks.h"
 #include "revng/Support/BasicBlockID.h"
 
 namespace yield::cfg {
 
-struct Node {
-  BasicBlockID Address;
-  BasicBlockID NextAddress;
-
-  explicit Node(BasicBlockID Address = BasicBlockID::invalid()) :
-    Address(std::move(Address)) {}
-  explicit Node(BasicBlockID Address, const MetaAddress &) :
-    Address(std::move(Address)) {}
-
-  bool isEmpty() const { return !Address.isValid(); }
+class Node {
+private:
+  std::string Location = {};
```

Comment:
Why store a string given that we seem to assume a specific location type (i.e., `revng::ranks::BasicBlock`)?

Reply:
Two reasons:
1. for consistency with the other node type, which can in fact point to two distinct location types.
2. because we're likely to want to extend this in the future with different types of nodes.

Reply:
1. Can we factor them out?
2. If so, add a comment.

Reply:
1. I tried to avoid unnecessary inheritance, there's no guarantee that we always want to keep unrelated node types linked. But if you want, it's not hard to separate common parts of the interface.

2. Sure thing.

Reply:
1. No, OK. I'm persuaded.
2. :+1:

Reply:
Works

Reply:
Ah, there's a typo in the second one. Keeping this open so I don't forget to fix it.

Reply:
Done.